### ISSUE TITLE
In window emit

### ISSUE THREAD

**Issue by chenziliang** [timeplus-io/proton#440]
**Use case**

Our community user likes to emit intermediate results before a window is closed and the community user liked to do per change emit.

We have discussed before regarding other in window emit strategy : periodical + window close which means before window close, still periodically emit window's result.

**Describe the solution you'd like**

```sql
-- In-window emit, every 1s emit aggregated results for a 5 mins window
SELECT window_start, user, count() 
FROM tumble(login, 5m) 
GROUP BY user, window_start
EMIT periodic 1s;
```

```sql
-- In-window emit, emit the ONLY window on every new data block on that window
SELECT window_start, user, count() 
FROM tumble(login, 5m) 
GROUP BY user, window_start
EMIT ON UPDATE;
```

**Describe alternatives you've considered**

**Additional context**

**Comment by chenziliang** [comment]
For 2) there may be multiple in-parallel windows, only emit the windows which have updates

**Comment by yl-lisen** [comment]
> For 2) there may be multiple in-parallel windows, only emit the windows which have updates

@chenziliang  I don't understand when there are multiple windows at the same time and only some of them are updated.
Do you means there are multiple groups, and only emit the groups which have updates.

**Comment by chenziliang** [comment]
Backfill without sorting.

I think we shall avoid sorting data for backfill. It is too slow and probably don’t worth it.

**Comment by yl-lisen** [comment]
> I think we shall avoid sorting data for backfill. It is too slow and probably don’t worth it.

Agree, in other hand, we should by default order by `_tp_time` instead of `to_start_of_hour(_tp_time)`, in my opinion they are the same effort

**Comment by yl-lisen** [comment]
> Backfill without sorting

We will filter late events via watermark, so it is not possible to handle multiple windows out of order

**Comment by chenziliang** [comment]
During backfill, skip stamping watermark, but only allow flush emit flag ?

**Comment by yl-lisen** [comment]
A behavior changes about: emit periodic 2s ,
currently, emit aggr results per 2s when there are updates.
I want to refine it into two modes:
emit periodic 2s  - emit aggr results per 2 seconds
emit periodic 2s on update  - emit aggr results per 2 seconds when there are updates. (omitted mode)

 for default behavior and user usage (syntax), any suggests? @chenziliang @jovezhong @gangtao

**Comment by jovezhong** [comment]
(Jove Github Bot) assuming it is not done, deferred this ticket to the next sprint.

**Comment by jovezhong** [comment]
(Jove Github Bot) assuming it is not done, deferred this ticket to the next sprint.

**Comment by jovezhong** [comment]
(Jove Github Bot) assuming it is not done, deferred this ticket to the next sprint.

**Comment by yl-lisen** [comment]
closed by #517
